src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -607,6 +623,103 @@ class HttpGoogleServicesDAO(
+      }).recover {
+        case t: HttpResponseException if StatusCode.int2StatusCode(t.getStatusCode) == StatusCodes.Conflict =>
+          throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"A google project by the name $projectName already exists"))
+      }
+
+      // set the billing account
+      billing <- retryWhen500orGoogleError(() => {
+        executeGoogleRequest(billingManager.projects().updateBillingInfo(projectResourceName, new ProjectBillingInfo().setBillingEnabled(true).setBillingAccountName(billingAccount.value)))
+      })
+
+      // get current permissions
+      bindings <- retryWhen500orGoogleError(() => {
+        executeGoogleRequest(cloudResManager.projects().getIamPolicy(projectName.value, null)).getBindings
+      })
+
+      // add any missing permissions
+      policy <- retryWhen500orGoogleError(() => {
+        val updatedPolicy = new Policy().setBindings(updateBindings(bindings, projectTemplate))
+        executeGoogleRequest(cloudResManager.projects().setIamPolicy(projectName.value, new SetIamPolicyRequest().setPolicy(updatedPolicy)))
+      })
+
+      // enable appropriate google apis
+      _ <- Future.sequence(projectTemplate.services.map { service => retryExponentially(when500orGoogleError)(() => {
+        enableServiceApi(projectName.value, service, credential)
+      })})
+
+      // create project usage export bucket
+      bucket <- retryWhen500orGoogleError(() => {
+        val bucket = new Bucket().setName(s"${projectName.value}-usage-export")
+        executeGoogleRequest(getStorage(credential).buckets.insert(projectName.value, bucket))
+      })
+
+      // set usage export bucket on project, it may take up to 5 minutes for the project to be ready for this but google is working to fix that
+      _ <- retryUntilSuccessOrTimeout(always)(5 seconds, 6 minutes)(() => {
+        Future(blocking(executeGoogleRequest(computeManager.projects().setUsageExportBucket(projectName.value, new UsageExportLocation().setBucketName(bucket.getName).setReportNamePrefix("usage")))))
+      })
+
+    } yield {
+      // nothing
+    }
+  }
+
+  def getComputeManager(credential: Credential): Compute = {
+    new Compute.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def getCloudBillingManager(credential: Credential): Cloudbilling = {
+    new Cloudbilling.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def getCloudResourceManager(credential: Credential): CloudResourceManager = {
+    new CloudResourceManager.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  private def enableServiceApi(projectName: String, service: String, credential: Credential): Future[HttpResponse] = {
+    import spray.json._
+    import GoogleRequestJsonSupport._
+
+    // note that I could not find a google client library for this end point but I know how to http
```

Comment:
i wonder if there is a way to inject this url and payload into an object that can be passed through executeGoogleRequest? That way the logging code doesn't need to be duplicated every time we have a special case like this